### PR TITLE
update docs for async insert deduplication

### PR DESCRIPTION
**Pull request by hanfei1991**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: docs/en/operations/settings/merge-tree-settings.md [hunk 1]
```diff
@@ -176,6 +176,59 @@ Similar to [replicated_deduplication_window](#replicated-deduplication-window),
...
+The number of most recently async inserted blocks for which ClickHouse Keeper stores hash sums to check for duplicates.
+
+Possible values:
+
+-   Any positive integer.
+-   0 (disable deduplication for async_inserts)
+
+Default value: 10000.
+
+The [Async Insert](./settings.md#async-insert) command will be cached in one or more blocks (parts). For [insert deduplication](../../engines/table-engines/mergetree-family/replication.md), when writing into replicated tables, ClickHouse writes the hash sums of each inserts into ClickHouse Keeper. Hash sums are stored only for the most recent `replicated_deduplication_window_for_async_inserts` blocks. The oldest hash sums are removed from ClickHouse Keeper.
```
[the remark this answers is not in the document]
Comment (DanRoscigno) [reply docs/en/operations/settings/merge-tree-settings.md:190]: reading further, I am not sure if my suggestion is correct.  Is there a hash sum for each insert, or each block of inserts?  If each insert, then just reject this suggestion and change `inserts` to `insert` singular.

File: docs/en/operations/settings/merge-tree-settings.md [hunk 2]
```diff
@@ -176,6 +176,59 @@ Similar to [replicated_deduplication_window](#replicated-deduplication-window),
...
+
+The minimum interval (in milliseconds) to update the `use_async_block_ids_cache`
+
+Possible values:
+
+-   Any positive integer.
+
+Default value: 100.
+
+Normally, the `use_async_block_ids_cache` updates as soon as there're updates in the watching keeper path. However, the cache updates might be too frequent and become a heavy burden. This minimun interval prevent the cache from updating too fast. Note that if we set this value too long, the block with duplicated inserts will have a longer retry time.
```
Comment (DanRoscigno) [inline docs/en/operations/settings/merge-tree-settings.md:230]:
```suggestion
Normally, the `use_async_block_ids_cache` updates as soon as there are updates in the watching keeper path. However, the cache updates might be too frequent and become a heavy burden. This minimum interval prevents the cache from updating too fast. Note that if we set this value too long, the block with duplicated inserts will have a longer retry time. 
```

File: docs/en/operations/settings/settings.md [hunk 3]
```diff
@@ -1394,6 +1394,22 @@ By default, blocks inserted into replicated tables by the `INSERT` statement are
...
+Enables or disables insert deduplication of `ASYNC INSERT` (for Replicated\* tables).
+
+Possible values:
+
+-   0 — Disabled.
+-   1 — Enabled.
+
+Default value: 1.
+
+By default, async inserted into replicated tables by the `INSERT` statement enabling [async_isnert](#async-insert) are deduplicated (see [Data Replication](../../engines/table-engines/mergetree-family/replication.md)).
```
Comment (DanRoscigno) [inline docs/en/operations/settings/settings.md:1408]:
```suggestion
By default, async inserts are inserted into replicated tables by the `INSERT` statement enabling [async_isnert](#async-insert) are deduplicated (see [Data Replication](../../engines/table-engines/mergetree-family/replication.md)).
```

File: docs/en/operations/settings/settings.md [hunk 4]
```diff
@@ -1394,6 +1394,22 @@ By default, blocks inserted into replicated tables by the `INSERT` statement are
...
+For the replicated tables by default the only 10000 of the most recent inserts for each partition are deduplicated (see [replicated_deduplication_window_for_async_inserts](merge-tree-settings.md/#replicated-deduplication-window-async-inserts), [replicated_deduplication_window_seconds_for_async_inserts](merge-tree-settings.md/#replicated-deduplication-window-seconds-async-inserts)).
```
Comment (DanRoscigno) [inline docs/en/operations/settings/settings.md:1409]:
```suggestion
For the replicated tables, by default, only 10000 of the most recent inserts for each partition are deduplicated (see [replicated_deduplication_window_for_async_inserts](merge-tree-settings.md/#replicated-deduplication-window-async-inserts), [replicated_deduplication_window_seconds_for_async_inserts](merge-tree-settings.md/#replicated-deduplication-window-seconds-async-inserts)).
```

**Review by DanRoscigno — APPROVED** [review]
Thanks so much for writing this.  I am not sure if my first suggestion is correct.

File: docs/en/operations/settings/merge-tree-settings.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hanfei1991) [reply docs/en/operations/settings/merge-tree-settings.md:190]: it's each insert, sorry for bad gramma

[the remark this answers is not in the document]
Comment (DanRoscigno) [reply docs/en/operations/settings/merge-tree-settings.md]: Don't apologize for grammar until I start apologizing for my bad C++ :)

**Comment by DanRoscigno** [comment]
Thanks again!!!

**Pull request merged**